hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -554,14 +548,8 @@ private Map<String, List<HoodieRecord<HoodieMetadataPayload>>> fetchBaseFileAllR
   }
 
   private HoodieRecord<HoodieMetadataPayload> composeRecord(GenericRecord avroRecord, String partitionName) {
-    if (metadataTableConfig.populateMetaFields()) {
-      return SpillableMapUtils.convertToHoodieRecordPayload(avroRecord,
-          metadataTableConfig.getPayloadClass(), metadataTableConfig.getPreCombineField(), false);
-    }
-    return SpillableMapUtils.convertToHoodieRecordPayload(avroRecord,
-        metadataTableConfig.getPayloadClass(), metadataTableConfig.getPreCombineField(),
-        Pair.of(metadataTableConfig.getRecordKeyFieldProp(), metadataTableConfig.getPartitionFieldProp()),
-        false, Option.of(partitionName), Option.empty());
+    return new HoodieAvroRecord<>(new HoodieKey(avroRecord.get(HoodieMetadataPayload.KEY_FIELD_NAME).toString(), partitionName),
```

Comment:
yeah, the record key is constant `key` now, if would be better to fetch it from table config in case this could change in future?

Reply:
I can update this but there are other places using this hardcoded approach. Changing the key of the metadata table would be a large breaking change.